core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -167,7 +168,8 @@ protected CqlRequestHandler(
     this.sessionMetricUpdater = session.getMetricUpdater();
 
     this.timer = context.getNettyOptions().getTimer();
-    Duration timeout = Conversions.resolveRequestTimeout(statement, context);
+    this.executionProfile = Conversions.resolveExecutionProfile(initialStatement, context);
+    Duration timeout = Conversions.resolveRequestTimeout(statement, executionProfile);
```

Comment:
This change does introduce a change in behaviour that I think is worth calling out and talking about.  I don't think it amounts to a substantial change so I suspect it'll be just fine... but it did seem worth discussing.

The current impl of 4.x resolves execution profiles in real time.  Indeed that's kind of the point of this entire change; we want to do this resolution a lot less.  The effect of resolving these profiles in real time means that any changes we might make to the configuration (say via the driver's support for [manual config reloading](https://github.com/apache/cassandra-java-driver/tree/4.x/manual/core/configuration#manual-reloading)) between sending the message (at the creation of CqlRequestHandler) and receipt of the response (via the callback) will be reflected in the callback handling.  That is no longer the case with this change; since we resolve a single ExecutionProfile at CqlRequestHandler creation and call it directly in the callback any configuration changes won't be reflected in the behaviour of the callback.

To be very clear: I'm not saying this is a problem.  There's a decent argument to be made that an entire request/response cycle should have a constant config in play, something which wasn't present before and is now.  So there's a non-trivial argument that this approach is actually _better_.  Either way it is a _difference_, however, so it seemed worthwhile spending a moment on it.

Reply:
> That is no longer the case with this change

Sure. I'm expecting request to be short lived (5 second ish) and assumed that we only needed the configuration to eventually be propagated.

Reply:
> To be very clear: I'm not saying this is a problem. There's a decent argument to be made that an entire request/response cycle should have a constant config in play, something which wasn't present before and is now. So there's a non-trivial argument that this approach is actually better. Either way it is a difference, however, so it seemed worthwhile spending a moment on it.

:+1:, I think it is good to call out, and that this change is more desirable (on the fly changes should only apply to subsequent requests, not in flight ones), but in any case I don't think it good to rely on the behavior working either way.